Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIMenusCocoa.mm

```diff
@@ -32,36 +32,131 @@
 
 #if ENABLE(WK_WEB_EXTENSIONS)
 
+#import "WebExtensionContextProxyMessages.h"
+#import "WebExtensionMenuItem.h"
+#import "WebExtensionMenuItemContextParameters.h"
+#import "WebExtensionMenuItemParameters.h"
+#import "WebExtensionUtilities.h"
+
 namespace WebKit {
 
+static bool isAncestorOrSelf(WebExtensionContext& context, const String& potentialAncestorIdentifier, const String& identifier)
+{
+    if (potentialAncestorIdentifier == identifier)
+        return true;
+
+    RefPtr current = context.menuItem(identifier);
```

Comment:
Does auto work here?

Reply:
Yes, but `context.menuItem` returns a pointer, so this protects it while in scope.

Also even if `RefPtr` or `Ref` is returned, I've been told by @cdumez that we should be explicit and use `RefPtr` or `Ref` instead of `auto` as WebKit is adopting smart pointer aggressively and that makes the calcite clear to anyone doing a code audit or review.